### PR TITLE
Update Options on Default Handler

### LINKED ISSUE #12: Not Updating JSON Field
Currently I'm developing apps that require one of the field to be the type of JSON. [(Prisma Reference)](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields)

The problem is every time the data is updated to the database, the JSON field would not be updated. I've done some digging and found this code at the file `updateHandler.ts`.

```ts
      if (!isObject(value) && !options?.skipFields?.includes(key))
        fields[key] = value;

      return fields;
```
[Link to code](https://github.com/codeledge/ra-data-simple-prisma/blob/cd20a0bffeefd5bf0b7ef3e4a0eee65b28508815/packages/ra-data-simple-prisma/src/updateHandler.ts#L15)

I think the problem is, that this library thinks the JSON field is actually a relation to another model, so it is skipped. I think there should be a way to differentiate it, or at least have a configuration to allow a particular field on a model to be updated.

### PR DESCRIPTION
**Pull request by fauh45**
- Change the layout of the default handler options
- Add allowFields to allow certain fields to not be checked by the "relations checker"

Example of usage

```ts
defaultHandler(req, res, prisma, {
    update: { allowFields: ["content"] },
});
```

With the configuration above, the field `content` will be allowed to be updated even thought it might be an object. This is particularly useful for those who uses JSON fields in their Prisma schema.

Downside is, whatever the model might be, if there are a same field name in one or more model, it will all be allowed and not be checked. So it does need to be used carefully. In the future the options should include the model name, and the fields.

Closes #12

### CODE REVIEW

File: packages/ra-data-simple-prisma/src/defaultHandler.ts [hunk 1]
```diff
@@ -56,22 +59,27 @@ export const defaultHandler = async (
       return await createHandler(req as CreateRequest, res, prismaDelegate);
     }
     case "update": {
-      return await updateHandler(req as UpdateRequest, res, prismaDelegate);
+      return await updateHandler(
+        req as UpdateRequest,
+        res,
+        prismaDelegate,
+        options.update
```
Comment (ogroppo) [inline packages/ra-data-simple-prisma/src/defaultHandler.ts]: believe this also needs a `options?.update`

**Review by ogroppo — COMMENTED** [review]
Looks good! just that little change please

File: packages/ra-data-simple-prisma/src/defaultHandler.ts (hunk 1 above)
Comment (fauh45) [reply packages/ra-data-simple-prisma/src/defaultHandler.ts]: Ah right, changing it right away.

**Comment by fauh45** [comment]
I believe the issue has been resolved. Thanks for the check, that was somehow gone past the typescript linter.

**Comment by ogroppo** [comment]
Cool, are you comfortable in adding a json field in the example app? It requires a prisma migration in `Post` model and installing `https://github.com/MrHertal/react-admin-json-view`... But if you want to push this back no worries I can merge and publish the package!

**Comment by ogroppo** [comment]
Cool, are you comfortable in adding a json field in the example app? It requires a prisma migration in `Post` model and installing `https://github.com/MrHertal/react-admin-json-view`... But if you want to push this back no worries I can merge and publish the package!

**Comment by ogroppo** [comment]
Cool, are you comfortable in adding a json field in the example app? It requires a prisma migration in `Post` model and installing `https://github.com/MrHertal/react-admin-json-view`... But if you want to push this back no worries I can merge and publish the package!

**Comment by fauh45** [comment]
> Cool, are you comfortable in adding a json field in the example app? It requires a prisma migration in `Post` model and installing `https://github.com/MrHertal/react-admin-json-view`... But if you want to push this back no worries I can merge and publish the package!

Isn't the example app uses sqlite and json field is not supported there? [(Prisma Reference)](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#default-type-mappings-7)

**Comment by ogroppo** [comment]
Hah lol, you're right, we should move away from sqlite then

**Comment by ogroppo** [comment]
published ra-data-simple-prisma@0.2.11

**Comment by fauh45** [comment]
Nicee, thanks all!

**Pull request merged**